test/PurchaseItem.test.js

```diff
@@ -0,0 +1,279 @@
+import SailthruMobile from "../index";
+
+describe("PurchaseItem", () => {
+  var purchaseItem;
+
+  beforeEach(() => {
+    purchaseItem = new SailthruMobile.PurchaseItem(
+      2,
+      "myTitle",
+      3456,
+      "id",
+      "url"
+    );
+  });
+
+  describe("when created", () => {
+    it("should set the quantity", () => {
+      expect(purchaseItem.qty).toEqual(2);
+    });
+
+    it("should set the title", () => {
+      expect(purchaseItem.title).toEqual("myTitle");
+    });
+
+    it("should set the price", () => {
+      expect(purchaseItem.price).toEqual(3456);
+    });
+
+    it("should set the ID", () => {
+      expect(purchaseItem.id).toEqual("id");
+    });
+
+    it("should set the URL", () => {
+      expect(purchaseItem.url).toEqual("url");
+    });
+
+    describe("when quantity is wrong type", () => {
+      it("should throw a type error", () => {
+        const run = () => {
+          purchaseItem = new SailthruMobile.PurchaseItem(
+            "2",
+            "myTitle",
+            3456,
+            "id",
+            "url"
+          );
+        };
+        expect(run).toThrow(TypeError);
+      });
+    });
+
+    describe("when quantity is not an integer", () => {
+      it("should throw a type error", () => {
+        const run = () => {
+          purchaseItem = new SailthruMobile.PurchaseItem(
+            2.5,
+            "myTitle",
+            3456,
+            "id",
+            "url"
+          );
+        };
+        expect(run).toThrow(TypeError);
+      });
+    });
+
+    describe("when title is not a string", () => {
+      it("should throw a type error", () => {
+        const run = () => {
+          purchaseItem = new SailthruMobile.PurchaseItem(
+            2,
+            55,
+            3456,
+            "id",
+            "url"
+          );
+        };
+        expect(run).toThrow(TypeError);
```

Comment:
I think it would be a bit more beneficial if we could test actual error messages. But it's not a blocker!